### ISSUE TITLE
Models apparently have reserved keywords

### ISSUE THREAD

**Issue by nsossonko** [phalcon/cphalcon#11253]
The following keywords (when used as column names) were found to cause issues with phalcon's models (although I could not find a list in the docs). IMHO they should not cause issues and should be allowed to be used, but at least it should be documented:

`group` - causes an exception during PHQL parsing as it sees it as meaning GROUP.
`transaction` - meddles with transactions used when nesting models.
`for` - Causes error similar to `group`'s.
`source` - Confuses phalcon to think we are declaring the table source.

Edit: I should note, this is using mysql backend.

**Comment by sergeyklay** [comment]
@nsossonko Thanks for reporting. Can you please provide a small script to reproduce? Also please provide Phalcon & MySQL version. We'll try to deal

**Comment by nsossonko** [comment]
Approximation of what we were doing:

```php
class Parents {
  $id;
  public function initialize() {
    $this->hasMany(
      'id',
      'Childs',
      'parent',
      array(
        'alias'  => 'children'
      )
    );
  }
}

class Childs {
  $id;
  $parent;
  $source;
  $transaction;
  $for;
  $group;
  public function initialize() {
    $this->belongsTo(
      'parent',
      'Parents',
      'id',
      array(
        'alias'  => 'myparent'
      )
    );
  }
}

$parent = new Parents();
$child = new Childs();
$child->myparent = $parent;
$child->transaction = 'transaction';
$child->create();
/**
 * Issue with transaction...you'll have to debug because, IIRC,
 * it was a difficult one to figure out
 */

$child = new Childs();
$child->source = 'source';
$child->create(); // Error about no such table

$children = Childs::findByFor(1); // Error
$children = Childs::findByGroup(1); // Error
```

SQL:

```sql
create table `parents` (
  `id` int not null auto_increment,
  primary key (`id`)
);

create table `childs` (
  `id` int not null auto_increment,
  `parent` int default null,
  `source` varchar(20) default null,
  `transaction` varchar(20) default null,
  `for` int default null,
  `group` int default null,
  primary key (`id`)
);
```

**Comment by Jurigag** [comment]
I guess in columnmap you have to put transaction etc in backticks ?

Its not really a phalcon problem:
http://dev.mysql.com/doc/refman/5.7/en/keywords.html

**Comment by nsossonko** [comment]
It is a Phalcon problem, because it fails in Phalcon code, not in PDO or Mysql. Phalcon already escapes everything with backticks so this wouldn't be a problem at all on the Mysql side. It fails in these scenarios as described above.

**Comment by Jurigag** [comment]
So show us logs, cuz we dont really now whats error it is.

**Comment by nsossonko** [comment]
So I've subsequently found this: https://docs.phalconphp.com/en/latest/reference/phql.html#escaping-reserved-words which doesn't detail the actual reserved words for phql, but makes enough sense I guess. That would fix `group` and `for`. `source` and `transaction` though are still very much an issue. `source` is very easy to test, just make a model with an attribute named `source` (as indicated in the code samples) and you will see it will attempt to use the `source` attribute's value as the table name. `transaction` is also demoed in the code above but is a bit more involved to test.

**Comment by sergeyklay** [comment]
Fixed in `2.1.x` branch
